contract/tsconfig.json

```diff
@@ -0,0 +1,11 @@
+{
+  "compilerOptions": {
+    "target": "es2020",
+    "module": "commonjs",
```

Comment:
let's move to esnext/esnext and make package a module in package.json?

```
      "target": "esnext",
      "module": "esnext",
      "moduleResolution": "node",
```

Reply:
Done

Reply:
I tried to do following your comment, but unable to do it.
Hardhat produces many issues when changing `package.json` like that, which I can't resolve. ;D
Will do it later.